Cython/Utility/StringTools.c

```diff
@@ -296,13 +302,23 @@ static CYTHON_INLINE int __Pyx_GetItemInt_ByteArray_Fast(PyObject* string, Py_ss
         #endif
         if (wraparound & unlikely(i < 0)) i += length;
         if ((!boundscheck) || likely(__Pyx_is_valid_index(i, length))) {
+            #if !CYTHON_ASSUME_SAFE_MACROS
+            char *asString = PyByteArray_AsString(string);
+            return (unsigned char)(likely(asString) ? asString[i] : -1);
```

Comment:
The cast seems wrong in the `-1` case. The function returns an `int`.
```suggestion
            return likely(asString) ? (unsigned char) asString[i] : -1;
```